src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

```diff
@@ -0,0 +1,692 @@
+						int[] selectedRows = jTableInventory.getSelectedRows();
+						if (selectedRows.length == 1) {
+							int selectedRow = jTableInventory.getSelectedRow();
+							MedicalInventory inventory = inventoryList.get(selectedRow);
+							if (inventory.getStatus().equals(InventoryStatus.canceled.toString()) ||
+								inventory.getStatus().equals(InventoryStatus.done.toString())) {
+								jButtonEdit.setEnabled(false);
+								jButtonDelete.setEnabled(false);
+							} else {
+								jButtonEdit.setEnabled(true);
+								jButtonDelete.setEnabled(true);
+
+							}
+							jButtonView.setEnabled(true);
+							jButtonView.setEnabled(true);
+							jButtonDelete.setEnabled(true);
+						} else {
+							jButtonEdit.setEnabled(false);
+							jButtonView.setEnabled(false);
+							jButtonDelete.setEnabled(false);
+						}
+					}
+
+				}
+			});
+		}
+		return jTableInventory;
+	}
+
+	class InventoryBrowsingModel extends DefaultTableModel {
+
+		private static final long serialVersionUID = 1L;
+
+		public InventoryBrowsingModel(int page, int pageSize) {
+			inventoryList = new ArrayList<>();
+			String state = statusComboBox.getSelectedIndex() > 0 ? statusComboBox.getSelectedItem().toString().toLowerCase() : null;
+			String type = InventoryType.main.toString();
+			try {
+				Page<MedicalInventory> medInventorypage = medicalInventoryManager.getMedicalInventoryByParamsPageable(dateFrom, dateTo, state, type, page,
+					pageSize);
+				inventoryList = medInventorypage.getContent();
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+		}
+
+		public Class< ? > getColumnClass(int c) {
+			if (c == 0) {
+				return String.class;
+			} else if (c == 1) {
+				return String.class;
+			} else if (c == 2) {
+				return String.class;
+			} else if (c == 3) {
+				return String.class;
+			}
+			return null;
+		}
+
+		public int getRowCount() {
+			if (inventoryList == null) {
+				return 0;
+			}
+			return inventoryList.size();
+		}
+
+		public String getColumnName(int c) {
+			return columsNames[c];
+		}
+
+		public int getColumnCount() {
+			return columsNames.length;
+		}
+
+		public Object getValueAt(int r, int c) {
+			MedicalInventory medInvt = inventoryList.get(r);
+			if (c == -1) {
+				return medInvt;
+			} else if (c == 0) {
+				return medInvt.getInventoryReference();
+			} else if (c == 1) {
+				return medInvt.getInventoryDate().format(DATE_TIME_FORMATTER);
+			} else if (c == 2) {
+				return medInvt.getStatus();
+			} else if (c == 3) {
+				return medInvt.getUser();
+			}
+			return null;
+		}
+
+		@Override
+		public boolean isCellEditable(int arg0, int arg1) {
+			return false;
+		}
+
+	}
+
+	public String formatDateTime(GregorianCalendar time) {
+		SimpleDateFormat format = new SimpleDateFormat("dd/MM/yy"); //$NON-NLS-1$
+		return format.format(time.getTime());
+	}
+
+	private JComboBox<String> getComboBox() {
+		if (statusComboBox == null) {
+			statusComboBox = new JComboBox<String>();
```

Comment:
```suggestion
			statusComboBox = new JComboBox<>();
```
use diamond operator